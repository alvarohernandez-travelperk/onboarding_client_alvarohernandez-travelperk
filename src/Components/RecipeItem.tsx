export default function RecipeItem() {

  const deleteRecipe = () => {
    console.log("Send delete request with id")
  };

  return (
    <div>
      <span> Recipe Title</span>
      <span> Recipe Description</span>
      <ul>
        <li>Ingredient 1</li>
        <li>Ingredient 2</li>
        <li>Ingredient 3</li>
      </ul>
      <button>Edit</button>
      <button onClick={()=>deleteRecipe()}>Delete</button>
    </div>
  );
}