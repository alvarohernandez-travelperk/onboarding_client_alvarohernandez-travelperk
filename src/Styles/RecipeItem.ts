import styled from 'styled-components'


export const Card = styled.div`
  padding: 20px;
  overflow: hidden;
  min-width: 300px;
  border: 1px solid #eaeaea;
  background-color: #fafafa;
  border-radius: 5px;
  margin: 16px 0px;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  width: 40%;
  @media only screen and (max-width: 600px) {
    padding: 10px;
    width: auto;
  }

  ul {
    list-style-type: none;
    padding: 20px;
    margin: 0px;

    li {
      padding: 2px;
    }
  }
`
export const Button = styled.button`
    border: 1px solid #0e1012;
    border-radius: 5px;
    height: 40px;
    width: 80px;
    padding: 0px 8px;
    cursor: pointer;
    background-color: #0e1012;
    color: #eaeaea;
    margin: 0px 10px;
    &:hover:enabled{
      color: #0e1012;
      border-color: #0e1012;
      background-color: #eaeaea;
    }
`

export const TextBox = styled.div`
  background-color: transparent;
  display: flex;
  margin: 10px;
`
export const IngredientsBox = styled.div`
  background-color: transparent;
  display: flex;
  margin: 10px;
  flex-direction: column;
`

export const Text = styled.p`
  color: #0e1012;
  background-color: transparent;
  text-decoration: none;
`
export const TextBold = styled.p`
  color: #0e1012;
  background-color: transparent;
  text-decoration: none;
  font-weight: 700;
  padding-right: 8px;
`