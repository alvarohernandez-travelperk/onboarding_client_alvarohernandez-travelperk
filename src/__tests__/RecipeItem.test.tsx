import { cleanup, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import RecipeItem from '../Components/RecipeItem';

const fetchData = jest.fn();
const mockRecipe = {
  title: "Mock Recipe",
  description: "Mock description",
  id: 0,
  ingredients: []
}

afterEach(cleanup);

it("Snapshot Test RecipeItem Component", () => {
  const { asFragment } = render(<RecipeItem
    recipe={mockRecipe}
    fetchData={fetchData} />,  {wrapper: BrowserRouter});
  expect(asFragment).toMatchSnapshot();
});

it("Test RecipeItem Texts", () => {
  const { getByTestId } = render(<RecipeItem
    recipe={mockRecipe}
    fetchData={fetchData} />,  {wrapper: BrowserRouter});
    expect(getByTestId("recipe-item-title")).toHaveTextContent(mockRecipe.title);
    expect(getByTestId("recipe-item-description")).toHaveTextContent(mockRecipe.description);
});