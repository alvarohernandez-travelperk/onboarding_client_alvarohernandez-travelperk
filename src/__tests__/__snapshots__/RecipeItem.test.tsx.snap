// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test RecipeItem Component 1`] = `[Function]`;
